test/terraform_aws_s3_private_bucket_test.go

```diff
@@ -373,3 +373,78 @@ func TestTerraformAwsS3PrivateBucketNoLoggingBucket(t *testing.T) {
 	AssertS3BucketLoggingNotEnabled(t, awsRegion, testName)
 	AssertS3BucketPolicyContainsNonTLSDeny(t, awsRegion, testName)
 }
+
+func AssertS3BucketAnalyticsEnabled(t *testing.T, region string, bucketName string) {
+	err := AssertS3BucketAnalyticsEnabledE(t, region, bucketName)
+	require.NoError(t, err)
+}
+
+func AssertS3BucketAnalyticsEnabledE(t *testing.T, region string, bucketName string) error {
+	s3Client, err := aws.NewS3ClientE(t, region)
+
+	if err != nil {
+		return err
+	}
+
+	params := &s3.ListBucketAnalyticsConfigurationsInput{
+		Bucket: awssdk.String(bucketName),
+	}
+
+	bucketAnalytics, err := s3Client.ListBucketAnalyticsConfigurations(params)
+
+	if err != nil {
+		return err
+	}
+
+	analyticsEnabled := bucketAnalytics.IsTruncated
+
+	if analyticsEnabled == nil {
+		return fmt.Errorf("Analytics is not enabled")
+	}
+
+	return nil
+}
+
+func TestTerraformAwsS3PrivateBucketAnalyticsBucket(t *testing.T) {
+	t.Parallel()
+
+	// TODO: create & hook tests below to tempTestFolder /examples/analytics after logging module connected
+
+	tempTestFolder := test_structure.CopyTerraformFolderToTemp(t, "../", "examples/simple-no-logging")
+
+	// Give this S3 Bucket a unique ID for a name tag so we can distinguish it from any other Buckets provisioned
+	// in your AWS account
+	testName := fmt.Sprintf("terratest-aws-s3-private-bucket-no-analytics-%s", strings.ToLower(random.UniqueId()))
+	awsRegion := "us-west-2"
+
+	terraformOptions := &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: tempTestFolder,
+
+		// Variables to pass to our Terraform code using -var options
+		Vars: map[string]interface{}{
+			"test_name": testName,
+			"region":    awsRegion,
+		},
+
+		// Environment variables to set when running Terraform
+		EnvVars: map[string]string{
+			"AWS_DEFAULT_REGION": awsRegion,
+		},
+	}
+
+	defer terraform.Destroy(t, terraformOptions)
```

Comment:
you have a defer destroy initandapply then test twice in this test....

Reply:
ooh totally copypasta 🤦‍♀️